MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -2083,187 +2057,31 @@ static void signalAndUntrack(const MVKSwapchainSignaler& signaler) {
-
-	// If we have an identity swizzle, we're all good.
-	if (SWIZZLE_MATCHES(R, G, B, A)) {
-		return VK_SUCCESS;
-	}
-
-	if (mvkIsAnyFlagEnabled(_imageView->_usage, VK_IMAGE_USAGE_INPUT_ATTACHMENT_BIT | VK_IMAGE_USAGE_STORAGE_BIT)) {
-		// Vulkan forbids using image views with non-identity swizzles as storage images or attachments.
-		// Let's catch some cases which are essentially identity, but would still result in Metal restricting
-		// the resulting texture's usage.
-
-		switch (_mtlPixFmt) {
-			case MTLPixelFormatR8Unorm:
-#if !MVK_OS_SIMULATOR
-			case MTLPixelFormatR8Unorm_sRGB:
-#endif
-			case MTLPixelFormatR8Snorm:
-			case MTLPixelFormatR8Uint:
-			case MTLPixelFormatR8Sint:
-			case MTLPixelFormatR16Unorm:
-			case MTLPixelFormatR16Snorm:
-			case MTLPixelFormatR16Uint:
-			case MTLPixelFormatR16Sint:
-			case MTLPixelFormatR16Float:
-			case MTLPixelFormatR32Uint:
-			case MTLPixelFormatR32Sint:
-			case MTLPixelFormatR32Float:
-				if (SWIZZLE_MATCHES(R, ZERO, ZERO, ONE)) {
-					return VK_SUCCESS;
-				}
-				break;
-
-			case MTLPixelFormatRG8Unorm:
-#if !MVK_OS_SIMULATOR
-			case MTLPixelFormatRG8Unorm_sRGB:
-#endif
-			case MTLPixelFormatRG8Snorm:
-			case MTLPixelFormatRG8Uint:
-			case MTLPixelFormatRG8Sint:
-			case MTLPixelFormatRG16Unorm:
-			case MTLPixelFormatRG16Snorm:
-			case MTLPixelFormatRG16Uint:
-			case MTLPixelFormatRG16Sint:
-			case MTLPixelFormatRG16Float:
-			case MTLPixelFormatRG32Uint:
-			case MTLPixelFormatRG32Sint:
-			case MTLPixelFormatRG32Float:
-				if (SWIZZLE_MATCHES(R, G, ZERO, ONE)) {
-					return VK_SUCCESS;
-				}
-				break;
-
-			case MTLPixelFormatRG11B10Float:
-			case MTLPixelFormatRGB9E5Float:
-				if (SWIZZLE_MATCHES(R, G, B, ONE)) {
-					return VK_SUCCESS;
-				}
-				break;
-
-			default:
-				break;
+		if (_componentSwizzle.g == VK_COMPONENT_SWIZZLE_A) {
+			_componentSwizzle.g = VK_COMPONENT_SWIZZLE_ONE;
+		} else if (_componentSwizzle.g == VK_COMPONENT_SWIZZLE_G || _componentSwizzle.g == VK_COMPONENT_SWIZZLE_B || _componentSwizzle.g == VK_COMPONENT_SWIZZLE_IDENTITY) {
+			_componentSwizzle.g = VK_COMPONENT_SWIZZLE_ZERO;
 		}
-	}
-
-	if (!_imageView->_image->hasPixelFormatView(_planeIndex)) {
-		if (!enableSwizzling()) {
-			MVKAssert(0, "Image without PixelFormatView usage couldn't enable swizzling!");
+		if (_componentSwizzle.b == VK_COMPONENT_SWIZZLE_A) {
+			_componentSwizzle.b = VK_COMPONENT_SWIZZLE_ONE;
+		} else if (_componentSwizzle.b == VK_COMPONENT_SWIZZLE_G || _componentSwizzle.b == VK_COMPONENT_SWIZZLE_B || _componentSwizzle.b == VK_COMPONENT_SWIZZLE_IDENTITY) {
+			_componentSwizzle.b = VK_COMPONENT_SWIZZLE_ZERO;
+		}
+		if (_componentSwizzle.a == VK_COMPONENT_SWIZZLE_A || _componentSwizzle.a == VK_COMPONENT_SWIZZLE_IDENTITY) {
+			_componentSwizzle.a = VK_COMPONENT_SWIZZLE_ONE;
+		} else if (_componentSwizzle.a == VK_COMPONENT_SWIZZLE_G || _componentSwizzle.a == VK_COMPONENT_SWIZZLE_B ) {
+			_componentSwizzle.a = VK_COMPONENT_SWIZZLE_ZERO;
```

Comment:
Oh. Geez. Are these the missing depth components from my comment above? 

If so, what the hell, GitDiff? 🤨